lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs

```diff
@@ -0,0 +1,157 @@
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'declaration-property-value-keyword-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (value, property) =>
+		`Unexpected deprecated keyword "${value}" for property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/declaration-property-value-keyword-no-deprecated',
+	fixable: true,
+};
+
+const DEPRECATED_COLORS = {
+	activecaption: 'canvas',
+	appworkspace: 'canvas',
+	background: 'canvas',
+	inactivecaption: 'canvas',
+	infobackground: 'canvas',
+	menu: 'canvas',
+	scrollbar: 'canvas',
+	window: 'canvas',
+	activeborder: 'ButtonBorder',
+	inactiveborder: 'ButtonBorder',
+	threeddarkshadow: 'ButtonBorder',
+	threedhighlight: 'ButtonBorder',
+	threedlightshadow: 'ButtonBorder',
+	threedshadow: 'ButtonBorder',
+	windowframe: 'ButtonBorder',
+	captiontext: 'CanvasText',
+	infotext: 'CanvasText',
+	menutext: 'CanvasText',
+	windowtext: 'CanvasText',
+	buttonhighlight: 'ButtonFace',
+	buttonshadow: 'ButtonFace',
+	threedface: 'ButtonFace',
+	inactivecaptiontext: 'GrayText',
+};
+const COLOR_PROPERTIES = [...singleValueColorProperties.values()].map(
+	(value) => /** @type {[string, Record<String, string>]} */ ([value, DEPRECATED_COLORS]),
+);
+const PROPERTY_NAME_TO_KEYWORD = new Map([
+	[
+		'appearance',
+		/** @type {string | string[] | Record<string, string>} */ ({
+			button: 'auto',
+			checkbox: 'auto',
+			listbox: 'auto',
+			menulist: 'auto',
+			meter: 'auto',
+			'progress-bar': 'auto',
+			'push-button': 'auto',
+			radio: 'auto',
+			searchfield: 'auto',
+			'slider-horizontal': 'auto',
+			'square-button': 'auto',
+			textarea: 'auto',
+		}),
+	],
+	[
+		'image-rendering',
+		{
+			optimizeQuality: 'smooth',
+			optimizeSpeed: 'pixelated',
+		},
+	],
+	['overflow', { overlay: 'auto' }],
+	['text-justify', { distribute: 'inter-character' }],
+	['text-orientation', { 'sideways-right': 'sideways' }],
+	['user-select', { element: 'contain' }],
+	['zoom', { reset: '1' }],
+	['text-decoration-line', 'blink'],
+	['box-sizing', 'padding-box'],
+	['image-orientation', 'flip'],
+	['min-height', ['intrinsic', 'min-intrinsic']],
+	['min-width', ['intrinsic', 'min-intrinsic']],
+	['max-height', ['intrinsic', 'min-intrinsic']],
+	['max-width', ['intrinsic', 'min-intrinsic']],
+	['height', ['intrinsic', 'min-intrinsic']],
+	['width', ['intrinsic', 'min-intrinsic']],
+	['word-break', 'break-word'],
+	...COLOR_PROPERTIES,
+]);
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				optional: true,
+				actual: secondaryOptions,
+				possible: {
+					ignoreValues: [isString, isRegExp],
+				},
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			const declarationValue = decl.value;
```

Comment:
```suggestion
			const {prop, value} = decl;
```

Let's destructure the node properties to be consistent with how we do it in other newer rules.

Reply:
It was on purpose.
i.e. it would conflict with https://github.com/stylelint/stylelint/pull/8223/files#diff-b1b4ecce949c46a09fbb6d105055c3c68d1da28fceaa18e1837fd92a8c4b8a1dR116

Reply:
Can we rename some of the other variables to clarify some conditionals?

Destructuring `const {prop, value} = decl;` is a common pattern in our rules, so let's use that and adapt the other variables.

I've left some suggestions below.